Lib/fontTools/cffLib/specializer.py

```diff
@@ -26,26 +27,63 @@ def programToString(program):
 	return ' '.join(str(x) for x in program)
 
 
-def programToCommands(program):
+def programToCommands(program, deltasToSourceValues=None, num_regions=0):
 	"""Takes a T2CharString program list and returns list of commands.
 	Each command is a two-tuple of commandname,arg-list.  The commandname might
 	be empty string if no commandname shall be emitted (used for glyph width,
 	hintmask/cntrmask argument, as well as stray arguments at the end of the
-	program (¯\_(ツ)_/¯)."""
+	program. The latter happens in subroutines.
+	Note that each 'blend' operator and its argument list is  replaced by a
+	list of arguments in which each argument item is a list of source font
+	values.
+	'deltasToSourceValues' can be from varLib.varStore.VarDataItem."""
 
 	width = None
 	commands = []
 	stack = []
+	seen_blend = False
+	seen_op = False
+	if num_regions != 0:
+		num_sources = num_regions + 1
```

Comment:
I think you should set `num_sources` to some value, even if `num_regions` is zero, to reduce the risk of `num_sources` being undefined at line 65.

Reply:
Thanks! I agree that this will make code safer against future editing. 'num_sources' can't currently be used unless num_regions is > 0, but this could  be changed in the future.